### PR TITLE
fix handling of select in pgvector handler & [Integration Testing] Test the PGVector data integration

### LINKED ISSUE #7758: [Integration Testing] Test the PGVector data integration
### Short description of current behavior

The objective of this issue is to test the PGVector data integration with MindsDB.

## Resources

Here are some useful links:

1. Follow the [README.md](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/handlers/pgvector_handler/README.md) file. It stores all the commands that need to be tested.
2. Use the [CREATE ML_ENGINE](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/handlers/pgvector_handler/README.md) command to connect PGVector data integration to MindsDB that is listed in the README.md file.

## Steps to Follow

1. Test the integration as outlined in the Resources above.
2. Save the test results, including commands and their outputs, in the `PGVector _QA.md` file in a new GitHub repository in your account. [Here is a sample Manual_QA.md file.](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/handlers/mysql_handler/Manual_QA.md)
3. Create a PR in the [MindsDB repository](https://github.com/mindsdb/mindsdb) following these steps:
	- Go to the `/mindsdb/integrations/handlers/` folder and open the `Handlers_Manual_QA.md` file.
	- If all the tests passed, append this line: `PGVector data integration Test Passed: <link to the test results in your GitHub repository>`
	- If some tests failed, create an issue to report it and append this line: `PGVector data integration Test Failed: <link to the test results in your GitHub repository>, <link to the issue>`

## :loudspeaker: Additional rewards :trophy:

Each integration test brings 10+ points for SWAG and entry into the draw for a :point_down: :
-  :computer:  [Razer Blade 16 Laptop](https://www.razer.com/gaming-laptops/Razer-Blade-16/RZ09-0483SEJ3-R3U1)
- :shirt: :bear:  [MindsDB Swag](https://store.covver.io/mindsdb/collections/swag-collection)

:information_source: For more info check out https://mindsdb.com/hacktoberfest/ :point_left:

### LINKED ISSUE #7773: [Bug]: Error with 'NoneType' object has no attribute 'get' in PGVector Handler
### Short description of current behavior

I'm encountering an issue while running the following SQL query:

`SELECT * FROM pvec.items;`

The error message I receive is:

`Error [pgvector/pvec]: 'NoneType' object has no attribute 'get'`

It seems like there might be a bug within the PGVector handler

### Video or screenshots

image

### Expected behavior

_No response_

### How to reproduce the error

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by maulanaakbardj**
## Description

The bug in the `pgvector_handler.py` code occurred due to improper handling of filter conditions and a missing check for `None` values. These changes have resolved the bug and ensure that the code functions as expected, both when `conditions `are provided and when they are not. 
And Update `README.md` for pgvector handler

Fixes #7758
Fixes  #7773

## Type of change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 📄 This change requires a documentation update

## Additional Media:
Before : 

`SELECT * FROM pvec.items;`

The error message I receive is:

`Error [pgvector/pvec]: 'NoneType' object has no attribute 'get'`

image

After : 
`Query successfully`

### CODE REVIEW

**Review by ZoranPandovski — DISMISSED** [review]
Thanks. Why we have create_table and insert removed?

**Comment by maulanaakbardj** [comment]
@ZoranPandovski , I removed create_table and insert because these functions have parameters that must always be listed that need to be satisfied, CMIIW. If I include create_table and insert, an error like this occurs: 
```
Traceback (most recent call last):
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\api\http\namespaces\sql.py", line 40, in post
    result = mysql_proxy.process_query(query)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\utilities\profiler\profiler.py", line 138, in wrapper
    result = function(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\api\mysql\mysql_proxy\mysql_proxy.py", line 495, in process_query
    executor.query_execute(sql)
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\utilities\profiler\profiler.py", line 138, in wrapper
    result = function(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\api\mysql\mysql_proxy\executor\executor.py", line 116, in query_execute
    self.do_execute()
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\utilities\profiler\profiler.py", line 138, in wrapper
    result = function(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\api\mysql\mysql_proxy\executor\executor.py", line 213, in do_execute
    ret = self.command_executor.execute_command(self.query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\utilities\profiler\profiler.py", line 138, in wrapper
    result = function(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\api\mysql\mysql_proxy\executor\executor_commands.py", line 603, in execute_command
    query = SQLQuery(statement, session=self.session)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\api\mysql\mysql_proxy\classes\sql_query.py", line 463, in __init__
    self.execute_query()
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\api\mysql\mysql_proxy\classes\sql_query.py", line 685, in execute_query
    raise e
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\api\mysql\mysql_proxy\classes\sql_query.py", line 679, in execute_query
    data = self.execute_step(step, steps_data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\api\mysql\mysql_proxy\classes\sql_query.py", line 764, in execute_step
    data = self._fetch_dataframe_step(step, steps_data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\api\mysql\mysql_proxy\classes\sql_query.py", line 575, in _fetch_dataframe_step
    data, columns_info = dn.query(
                         ^^^^^^^^^
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\utilities\profiler\profiler.py", line 138, in wrapper
    result = function(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data\Riset\Mindsdb\Main-Github\mindsdb\mindsdb\api\mysql\mysql_proxy\datahub\datanodes\integration_datanode.py", line 150, in query
    raise DBHandlerException(msg) from e
mindsdb.api.mysql.mysql_proxy.datahub.datanodes.integration_datanode.DBHandlerException: [pgvector/pvec]: 4 columns passed, passed data had 1 columns
```

**Comment by maulanaakbardj** [comment]
And when I run this handler with a query in README.md without function, I don't get any errors so this is my ref

**Comment by maulanaakbardj** [comment]
@ZoranPandovski , I have updated my code without removing the create_table and insert and upon testing, the output is good

**Comment by maulanaakbardj** [comment]
> @ZoranPandovski , I have updated my code without removing the create_table and insert and upon testing, the output is good

@ZoranPandovski, can you please review it again

**Comment by maulanaakbardj** [comment]
@ZoranPandovski @martyna-mindsdb, is there a review update for my code fix?

**Comment by maulanaakbardj** [comment]
Update Handlers_Manual_QA.

Fixes #7758

Hi @chandrevdw31 , could you please review it too

**Review by martyna-mindsdb — APPROVED** [review]

**Comment by dusvyat** [comment]
@ea-rus - bringing this one to your attention as you are currently making changes to pgvector handler also

**Comment by ea-rus** [comment]
I run mindsdb on this branch and tried to create table using this query.
```
create table pvec.items (
SELECT m.embedding AS embeddings, t.review content FROM  mysql_demo_db.amazon_reviews t
  join openai_emb m
)
```
 It created only two columns in table. On [this PR](https://github.com/mindsdb/mindsdb/pull/8123) it also had `id` and `metadata` columns

**Comment by maulanaakbardj** [comment]
Thanks @ea-rus for your review. 
Yeah, I think on this branch have an issue with the table definition used in the `create_table` method, as it doesn't impose specific constraints or data types on the `id` and `metadata` columns. 
```
cur.execute(
    f"CREATE TABLE IF NOT EXISTS {table_name} (id text PRIMARY KEY, content text, embeddings vector, metadata jsonb)"
)
```
This code snippet defines the table structure, in my code branch the `id` and `metadata` columns are regular columns are regular columns with `text` data type. so when the INSERT operation is executed later in the code it can result in that column not appearing in the output when the INSERT operation is executed.

Ok I update `pgvector_handler.py` on my branch with the same as [this PR ](https://github.com/mindsdb/mindsdb/pull/8123)

image

**Comment by maulanaakbardj** [comment]
I think with the latest changes, this resolves issues #7758 and #7773

**Comment by chandrevdw31** [comment]
@maulanaakbardj please resolve the conflicts

**Comment by maulanaakbardj** [comment]
@chandrevdw31 , conflicts have been resolved

**Review by dusvyat — APPROVED** [review]

File: mindsdb/integrations/handlers/pgvector_handler/README.md [hunk 1]
```diff
@@ -62,19 +62,51 @@ WITH
...
-    (SELECT embedding AS embeddings
-        FROM (SELECT * FROM mysql_demo_db.demo_fda_context
-        LIMIT 3) AS d
-join  openai_emb);
+--- Public demo database
+CREATE DATABASE mysql_demo_db
+WITH ENGINE = "mysql",
+PARAMETERS = {
+   "user": "user",
+   "password": "MindsDBUser123!",
```
Comment (martyna-mindsdb) [inline mindsdb/integrations/handlers/pgvector_handler/README.md:70]: Let's change the values of host, user, and password to some placeholders.

Comment (maulanaakbardj) [reply mindsdb/integrations/handlers/pgvector_handler/README.md]: @martyna-mindsdb, I updated the README to remove the public demo database. Is it ok?

**Pull request merged**